src/scrapper/spiders/slot_spider.py

```diff
@@ -112,21 +113,43 @@ def classUnitRequests(self):
         for course_unit in self.course_units:
             yield Request(                
                 url="https://sigarra.up.pt/{}/pt/{}".format(course_unit[2], course_unit[1]),
-                meta={'course_unit_id': course_unit[0]},
-                callback=self.extractSchedule,
+                meta={'course_unit_id': course_unit[0], 'faculty': course_unit[2], 'schedule_url': course_unit[1]},
+                callback=self.getScheduleBlocks,
                 errback=self.func
             )
             
     def func(self, error):
         # # O scrapper não tem erros
         # print(error)
         return
-
-    def extractSchedule(self, response):
-        # Check if there is no schedule available
+    
+    def getScheduleBlocks(self, response):
         if response.xpath('//div[@id="erro"]/h2/text()').extract_first() == "Sem Resultados":
             yield None
 
+        week_blocks = list(set(
+            response.xpath('//td[@class="l sem-quebra"]//a/@href').getall() 
+                               + 
+            response.xpath('//td[@class="bloco-select sem-quebra"]//a/@href').getall()
+           ))
+        
+        if (len(week_blocks) == 0):
+            yield scrapy.http.Request(
+                url="https://sigarra.up.pt/{}/pt/{}".format(response.meta['faculty'], response.meta['schedule_url']),
+                meta=response.meta,
+                callback=self.extractSchedule,
+                errback=self.func
+            )
+        else:
+            for week_block in week_blocks:
+                yield scrapy.http.Request(
+                    url="https://sigarra.up.pt/{}/pt/{}".format(response.meta['faculty'], week_block),
+                    meta=response.meta,
+                    callback=self.extractSchedule,
+                    errback=self.func
+                )
```

Comment:
And this block appear to the same thing. 

The whole function `getScheduleBlocks` appears to be repeated in the two spiders. Shouldn't it be in a file accessed by both spiders and then the spiders would call that function?